### PR TITLE
[ci][7up/1] upgrade release tests to py39

### PR DESCRIPTION
**Pull request by can-anyscale**
Upgrade release tests to run with py39 by default. Merge the requirement files of py38 and py39. I also use the requirements_compiled.txt as a constraint while I'm here.

Some dependencies conflict with each other. I moved those to individual test dependencies.

Test:
- Release test run: https://buildkite.com/ray-project/release/builds/4945

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

**Review by matthewdeng — APPROVED** [review]

File: release/ray_release/byod/requirements_ml_byod_3.9.in [hunk 1]
```diff
@@ -1,31 +1,61 @@
...
+boto3
+cmake
+crc32c
+cupy-cuda113
 dataset
 datasets
 decord
 deepspeed>=0.12.3
-diffusers
+diffusers==0.12.1
```
Comment (aslonnie) [inline release/ray_release/byod/requirements_ml_byod_3.9.in:14]: why is this pinned to 0.12.1 now?

File: release/nightly_tests/dask_on_ray/large_scale_dask_on_ray_compute_template.yaml [hunk 2]
```diff
@@ -11,11 +10,11 @@ aws:
 
 head_node_type:
     name: head_node
-    instance_type: i3.8xlarge
+    instance_type: m6i.8xlarge
 
 worker_node_types:
     - name: worker_node
-      instance_type: i3.8xlarge
+      instance_type: m6i.8xlarge
```
Comment (aslonnie) [inline release/nightly_tests/dask_on_ray/large_scale_dask_on_ray_compute_template.yaml:17]: how is this change related to the py39-upgrade?

File: release/ray_release/byod/requirements_byod_3.9.in [hunk 3]
```diff
@@ -1 +1,39 @@
 # Python requirements to run release tests from anyscale byod (cpu type)
+
+-c requirements_compiled.txt
```
[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/byod/requirements_byod_3.9.in:3]: it doesn't need to; i just add this constraint while I'm here (which something we talked about doing a while back)

File: release/nightly_tests/dask_on_ray/large_scale_dask_on_ray_compute_template.yaml (hunk 2 above)
Comment (can-anyscale) [reply release/nightly_tests/dask_on_ray/large_scale_dask_on_ray_compute_template.yaml:17]: this is the smoke test version of this test; it has been broken for a while, i just fix it here so I can run test with it

File: release/ray_release/byod/requirements_ml_byod_3.9.in (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/byod/requirements_ml_byod_3.9.in:14]: for this test to pass https://github.com/ray-project/ray/blob/master/doc/source/serve/tutorials/stable-diffusion.md?plain=1#L9, it doesn't work with the latest version of diffusers

Comment (aslonnie) [reply release/ray_release/byod/requirements_ml_byod_3.9.in:14]: could you consult people that know more about the test if this is the right thing to do?

**Review by aslonnie — APPROVED** [review]
I am a bit concerned about removing 3.8 entirely before the final decision is made to stop supporting 3.8, but will leave that for you to judge.

File: release/ray_release/byod/requirements_ml_byod_3.9.in (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/byod/requirements_ml_byod_3.9.in:14]: CC: @edoakes , @GeneDer is it ok to pin diffusers=0.12.1 for stable diffusion test. For context, the doc tutorial also pins to the same version.

Comment (GeneDer) [reply release/ray_release/byod/requirements_ml_byod_3.9.in:14]: Given the doc also pined diffusers to 0.12.1, I think this is the right thing to do. Serve team can follow up to upgrade the api at a later time.

**Comment by can-anyscale** [comment]
let's leave it here a bit to confirm from product

**Comment by can-anyscale** [comment]
Ok, product already onboard and informed customers - it might take a while for us to know if the customers might protest so let's do this. We can add py38 back if necessary, that's way easier than removing it.
